web_poet/mixins.py

```diff
@@ -78,3 +81,34 @@ def urljoin(self, url: str) -> str:
         url and baseurl of the response"""
         # FIXME: move it to HttpResponse
         return urljoin(self.base_url, url)
+
+
+class _UrlMixin(abc.ABC):
+    """Common shortcut methods for working with the URL of HTTP requests and
+    responses.
+
+    This mixin could be used with Page Object base classes.
+
+    It requires the ``url`` attribute to be present.
```

Comment:
I wonder if it'd be better to just move the relevant methods and attributes to HttpResponse class, without creating a mixin. 

It seems there are 2 use cases for having a mixin instead of implementing necessary methods directly:

1. To use it for HttpRequest.
2. To use it as a mixin for page objects.

For HttpRequest there is no base_url, it seems the implementation of .urljoin can be pretty straightforward - just a shortcut to urljoin. It doesn't look like there is going to be code duplication.

For custom page objects the interface doesn't look straightforward - user needs to implement url and text properties. I think we can think about a better shortcut, if that's needed (e.g. a base class). A shortcut may proxy the call to self.response.urljoin, so there won't be duplication either.

Also, we can always add a mixin later, if it turns out there are more advantages to it. But currently for me it looks like moving this logic to HttpResponse class makes the code simpler, and removes these implicit "url" and "text" dependencies, as well as indirection.